From 723c6a1266785545f02844e800746ceeb9f3c54a Mon Sep 17 00:00:00 2001
From: Stefan Agner <stefan@agner.ch>
Date: Mon, 9 Dec 2019 13:06:36 +0100
Subject: [PATCH] backend-drm: Define potentially missing aspect-ratio bit
 definitions

Upstream-Status: Backport

The aspect ratio definitions for 64:27 and 256:135 have been added with
libdrm 2.4.95. However, Weston currently depends on libdrm 2.4.89 or
higher. Define the definitions in Weston to support libdrm older than
2.4.95.

Fixes: #332
Fixes: 6093772f45a4 ("backend-drm: Use aspect-ratio bit definitions from libdrm")
Signed-off-by: Stefan Agner <stefan@agner.ch>
---
 libweston/backend-drm/drm-internal.h | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/libweston/backend-drm/drm-internal.h b/libweston/backend-drm/drm-internal.h
index 975fa26..1848ad9 100644
--- a/libweston/backend-drm/drm-internal.h
+++ b/libweston/backend-drm/drm-internal.h
@@ -114,6 +114,18 @@
 
 #define MAX_CLONED_CONNECTORS 4
 
+#ifndef DRM_MODE_PICTURE_ASPECT_64_27
+#define DRM_MODE_PICTURE_ASPECT_64_27		3
+#define  DRM_MODE_FLAG_PIC_AR_64_27 \
+			(DRM_MODE_PICTURE_ASPECT_64_27<<19)
+#endif
+#ifndef DRM_MODE_PICTURE_ASPECT_256_135
+#define DRM_MODE_PICTURE_ASPECT_256_135		4
+#define  DRM_MODE_FLAG_PIC_AR_256_135 \
+			(DRM_MODE_PICTURE_ASPECT_256_135<<19)
+#endif
+
+
 /**
  * Represents the values of an enum-type KMS property
  */
-- 
2.7.4

